### PR TITLE
VIH-7907 quick link users in participants lists

### PR DESCRIPTION
**Pull request by FarronF**
### JIRA link (if applicable) ###
VIH-7907
VIH-8189

### Change description ###
Adding quick link users to count for judges hearing list. Removing party on tooltip in conference/consultation. Including QL observers in observers table. Ordering as per ticket for waiting room. Removing quick link observers from consultation lists.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts [hunk 1]
```diff
@@ -118,14 +118,23 @@ export abstract class WRParticipantStatusListDirective implements DoCheck {
...
+                x.hearing_role !== HearingRole.QUICK_LINK_PARTICIPANT &&
                 x.hearing_role !== HearingRole.STAFF_MEMBER
         );
 
+        nonJudgeParts = [
+            ...nonJudgeParts,
+            ...this.conference.participants
+                .filter(x => x.hearing_role === HearingRole.QUICK_LINK_PARTICIPANT)
+                .sort((a, b) => a.display_name.localeCompare(b.display_name))
+        ];
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:136]: Adding alphabetized quick link participants to the end of this list.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts [hunk 2]
```diff
@@ -186,7 +195,9 @@ export abstract class WRParticipantStatusListDirective implements DoCheck {
     }
 
     protected filterObservers(): void {
-        this.observers = this.conference.participants.filter(x => x.hearing_role === HearingRole.OBSERVER);
+        this.observers = this.conference.participants
+            .filter(x => x.hearing_role === HearingRole.OBSERVER || x.hearing_role === HearingRole.QUICK_LINK_OBSERVER)
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:199]: should we not use the role property out of the participants, I am suggesting this as the role appears to be an enum that is derived from the UserRole property of video api, where as this hearing role is just a string

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts [hunk 3]
```diff
@@ -119,7 +120,9 @@ export class PrivateConsultationParticipantsComponent extends WRParticipantStatu
 
     getPrivateConsultationParticipants(): ParticipantListItem[] {
         return this.participantsInConsultation
-            .filter(c => c.hearing_role !== HearingRole.WITNESS && c.hearing_role !== HearingRole.OBSERVER)
+            .filter(
+                c => c.hearing_role !== HearingRole.WITNESS && c.hearing_role !== HearingRole.OBSERVER && c.role !== Role.QuickLinkObserver
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:124]: both the hearing role and the role gets mapped in the backend of video web inside of ParticipantRequestMapper.cs, so I think it should be sufficient enough to check the role here, as hearing role itself is a string value and not sure how reliable that can be. Unless we are using hearing role here for some reason

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts (hunk 2 above)
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:199]: Essentially they should be the same, but yeah I did start prefering Role over HearingRole for QuickLink users. I will change this.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts (hunk 3 above)
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:124]: This does use role for QuickLinksObserver. We cannot use it for Witness or Observer though. Is that what you were suggesting?

Comment (omio-ma) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:124]: yea you are right, I misread the hearing role observer to be quick link hearing role observer, this is fine and I will mark it as resolved

**Review by omio-ma — APPROVED** [review]

**Comment by FarronF** [comment]
Changing to a draft as I discussed some changes with Lorenzo and we discovered that there is a bug in production that affected my implementation of this ticket. Will reopen when happy to merge again.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/panel-model-base.ts [hunk 4]
```diff
@@ -158,10 +158,12 @@ export abstract class PanelModel {
...
         } else if (this.role === Role.JudicialOfficeHolder) {
             return 2;
-        } else if (this.caseTypeGroup.toLowerCase() === 'endpoint') {
+        } else if (this.role === Role.QuickLinkParticipant) {
             return 4;
-        } else if (this.hearingRole === HearingRole.OBSERVER) {
+        } else if (this.caseTypeGroup.toLowerCase() === 'endpoint') {
             return 5;
+        } else if (this.hearingRole === HearingRole.OBSERVER || this.role === Role.QuickLinkObserver) {
+            return 6;
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/panel-model-base.ts:166]: This is for the main conference. Extending the way it currently works. Quick link participants go at the end of regular. And quicklink observers and regular observers are mixed

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.html [hunk 5]
```diff
@@ -57,5 +59,20 @@ <h2 class="govuk-heading-s white">{{ 'private-consultation-participants.particip
...
+                                    [interpreter]="participant?.interpreter"
+                                    [conferenceId]="conference.id"
+                                    [roomLabel]="roomLabel"
+                                    [participantCallStatuses]="participantCallStatuses"
+                                    [canInvite]="canInvite"
+                                    >
+                    </app-participant-item>
+                </div>
+            </ng-container>
+        </ng-container>
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.html:76]: For ticket VIH-8189 show observers and witness in JohConsulatation

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts [hunk 6]
```diff
@@ -118,12 +119,13 @@ export class PrivateConsultationParticipantsComponent extends WRParticipantStatu
...
     getPrivateConsultationParticipants(): ParticipantListItem[] {
-        return this.participantsInConsultation
+        const regularParticipants = this.participantsInConsultation
             .filter(
                 c =>
                     c.hearing_role !== HearingRole.WITNESS &&
                     c.hearing_role !== HearingRole.OBSERVER &&
-                    c.hearing_role !== HearingRole.QUICK_LINK_OBSERVER
+                    c.role !== Role.QuickLinkObserver &&
+                    c.role !== Role.QuickLinkParticipant
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:128]: Don't include initially as they should be sorted seperately

File: VideoWeb/VideoWeb.UnitTests/Helpers/ParticipantsUpdatedEventNotifierTests.cs [hunk 7]
```diff
@@ -58,7 +58,7 @@ public async Task Should_send_event()
 
 
             var response2 = new ParticipantResponse();
-            response2.Id = _participant1.Id;
+            response2.Id = _participant2.Id;
```
Comment (FarronF) [inline VideoWeb/VideoWeb.UnitTests/Helpers/ParticipantsUpdatedEventNotifierTests.cs:61]: This is a change to a different test that I thought was already in master.

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/models/JudgeHearingSummary.ts [hunk 8]
```diff
@@ -11,13 +11,15 @@ export class JudgeHearingSummary extends HearingSummary {
 
     get nonJudicialParticipantsExcludingObservers(): ParticipantSummary[] {
         const p = this.participants.filter(
-            x => (x.role === Role.Individual || x.role === Role.Representative) && x.hearingRole !== HearingRole.OBSERVER
+            x =>
+                (x.role === Role.Individual || x.role === Role.Representative || x.role === Role.QuickLinkParticipant) &&
```
Comment (krishnapottekkat-hmcts) [inline VideoWeb/VideoWeb/ClientApp/src/app/shared/models/JudgeHearingSummary.ts:15]: Can we change this to x.role === Role.QuickLinkObserver ? As the previous ticket VIH-8011 in QA is related to observer. I will bridge this as part of 7905 (QL Participant)

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/models/judge-hearing-summary.spec.ts [hunk 9]
```diff
@@ -65,13 +66,15 @@ describe('JudgeHearingSummary', () => {
...
 
         expect(
-            participants.filter(x => x.role !== Role.Individual && x.role !== Role.Representative && x.hearingRole !== HearingRole.WINGER)
-                .length
+            participants.filter(
+                x =>
+                    x.role !== Role.Individual &&
+                    x.role !== Role.Representative &&
+                    x.hearingRole !== HearingRole.WINGER &&
+                    x.role !== Role.QuickLinkParticipant
```
Comment (krishnapottekkat-hmcts) [inline VideoWeb/VideoWeb/ClientApp/src/app/shared/models/judge-hearing-summary.spec.ts:74]: QuickLinkObserver ?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts [hunk 10]
```diff
@@ -140,16 +142,31 @@ export class PrivateConsultationParticipantsComponent extends WRParticipantStatu
                 }
                 return participant;
             });
+        const quickLinkParticipants = this.participantsInConsultation
+            .filter(p => p.role === Role.QuickLinkParticipant)
```
Comment (krishnapottekkat-hmcts) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:146]: QuickLinkObserver ?

Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:146]: Quick link observers are not to be displayed in these lists except in joh consultation rooms. They are included in getWitnessesAndObservers.

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/models/JudgeHearingSummary.ts (hunk 8 above)
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/models/JudgeHearingSummary.ts:15]: This function is to exclude Observers and quick link observers. They are part of a separate table.

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/models/judge-hearing-summary.spec.ts (hunk 9 above)
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/models/judge-hearing-summary.spec.ts:74]: As mentioned in my other comment quicklink observers will not be included in this function.

**Review by krishnapottekkat-hmcts — APPROVED** [review]
LGTM

**Review by omio-ma — APPROVED** [review]

**Pull request merged**